### ISSUE TITLE
[BUG]Network does not allow downloading python binary from official website.

### ISSUE THREAD

**Issue by dinjazelena** [mlflow/mlflow#9804]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

mlflow==2.5.0

### System information

DBR 13.3 LTS ML

### Describe the problem

```
model = mlflow.pyfunc.spark_udf(spark=spark, model_uri="models:/test_mlflow_env_model/latest", env_manager="virtualenv")

So when logging model i added pip_requirements.

When loading as spark_udf i get following

ShellCommandException: Non-zero exit code: 1
Command: ['/databricks/.pyenv/bin/pyenv', 'install', '--skip-existing', '3.10.12']

STDERR:
Downloading Python-3.10.12.tar.xz...
-> https://www.python.org/ftp/python/3.10.12/Python-3.10.12.tar.xz
error: failed to download Python-3.10.12.tar.xz

BUILD FAILED (Ubuntu 22.04 using python-build 20180424)

Results logged to /tmp/python-build.20231003123449.4221.log

Last 10 log lines:
/tmp/python-build.20231003123449.4221 /databricks/driver
curl: (35) error:0A000152:SSL routines::unsafe legacy renegotiation disabled
```

Because i dont have network access for python website. How can i override this?

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [X] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@dinjazelena you can try `model = mlflow.pyfunc.spark_udf(spark=spark, model_uri="models:/test_mlflow_env_model/latest", env_manager="local")`

**Comment by dinjazelena** [comment]
@harupy , but thats the point, i dont want to use local environment, but environment that model was trained on.

**Comment by dinjazelena** [comment]
@harupy , yep, its just env manager tries to go to python org website to get python binary, but we have artifactory pypi proxy and no access to public github or python.org.
And i can redirect pypi to our pypi mirror, but this first step to download python binary, i cannot change it.

**Comment by dinjazelena** [comment]
@harupy 
```
python_exe = "/local_disk0/.ephemeral_nfs/envs/pythonEnv-c69f507c-73bd-4e2b-bc23-ac0bd7b3ed08/bin/python"

with mock.patch("mlflow.utils.virtualenv. _install_python", return_value=python_exe):

    model = mlflow.pyfunc.spark_udf(spark=spark, model_uri="models:/test_mlflow_env_model/latest", env_manager="virtualenv")
```

AttributeError: <module 'mlflow.utils.virtualenv' from '/databricks/python/lib/python3.10/site-packages/mlflow/utils/virtualenv.py'> does not have the attribute ' _install_python'

**Comment by harupy** [comment]
@dinjazelena You have space before `_install_python`.

**Comment by dinjazelena** [comment]
Shit, stupid me, thank u :)

**Comment by harupy** [comment]
@dinjazelena did it work?
